README.md

```diff
@@ -72,12 +65,40 @@ A valid (read: that follows this interface) IPLD format implementation the follo
 
 `callback` must have the signature `function (err, dagNode)`, where `err` is an Error if the function fails and `dagNode` is the dagNode that got deserialized in the process.
 
-#### `util.cid(dagNode, callback)`
+#### `util.cid(dagNode[, options], callback)`
 
 > get the CID of the dagNode
 
+Options include:
+  - version - the CID version to be used (defaults to 1)
+  - hashAlg - the hash algorithm to be used (default to the one set by the format)
+
 `callback` must have the signature `function (err, cid)`, where `err` is an Error is the function fails and `cid` is a CID instance of the dagNode.
 
+#### `util.cidSerialized(binaryBlob[, options], callback)`
+
+> get the CID of the serialized dagNode
+
+Options include:
+  - version - the CID version to be used (defaults to 1)
+  - hashAlg - the hash algorithm to be used (default to the one set by the format)
+
+`callback` must have the signature `function (err, cid)`, where `err` is an Error is the function fails and `cid` is a CID instance of the dagNode.
```

Comment:
@wanderer resisted swapping dagNode by serialized node and instead decided to create another method.

Reply:
At the block service level, we have the Block class that carries the CID there. However, at the DAG Level, the user should be able to specify its CID.

I believe we have the same :)

Reply:
not to build an anti-corrosion-bike-shelter - but the name would make me think this **produces** a cid that is serialized. Maybe cidFromBinary / cidFromBlob / addCidPrefix?

Reply:
@diasdavid I'd like to keep the surface of this API as little as possible. Higher level APIs should be part of js-ipld. Hence I'd prefer having a single `util.cid()` which works on the serialized node. If you want to get the CID of a DAG node, you could just call `serialize()` first, which you would need to call internally anyway, wouldn't you?

Reply:
@vmx 👍 from me.